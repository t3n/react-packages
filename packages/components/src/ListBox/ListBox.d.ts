import React, { ReactNode } from 'react';
import { LayoutProps } from 'styled-system';
export interface LayoutWithChildrenProps extends LayoutProps {
    children?: ReactNode;
}
declare const ListBox: React.FC<LayoutWithChildrenProps>;
export default ListBox;
