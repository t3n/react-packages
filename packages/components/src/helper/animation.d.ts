import { ThemeProps } from '@t3n/theme';
export declare const backgroundAnimation: import("styled-components").FlattenInterpolation<import("styled-components").ThemedStyledProps<ThemeProps, any>>;
