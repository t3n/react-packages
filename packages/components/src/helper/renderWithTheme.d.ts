import 'jest-styled-components';
import React from 'react';
import { RenderOptions } from '@testing-library/react';
import { Theme } from '@t3n/theme';
interface OptionalTheme {
    theme?: Theme;
}
export declare const renderWithTheme: (ui: React.ReactElement, { theme, ...options }: RenderOptions & OptionalTheme) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement>;
export {};
