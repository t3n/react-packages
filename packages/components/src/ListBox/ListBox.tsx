import React from 'react';
import styled from 'styled-components';
import { border, LayoutProps } from 'styled-system';

import Box from '../Box';

const Container = styled(Box)`
  ${({ theme }) =>
    border({
      theme,
      borderColor: 'shades.grey42',
      borderWidth: '1px 2px 4px 1px',
      borderStyle: 'solid',
      borderRadius: theme.border.radii[1],
    })}
`;

const ListBox: React.FC<LayoutProps> = ({ children, ...rest }) => {
  return (
    <Container {...rest} p={[3, 3, 5, 5]}>
      {children}
    </Container>
  );
};

export default ListBox;
