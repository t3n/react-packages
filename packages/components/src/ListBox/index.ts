import ListBox from './ListBox';

export * from './ListBox';

export default ListBox;
