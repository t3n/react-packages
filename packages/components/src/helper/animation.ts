import { css, keyframes } from 'styled-components';

import { ThemeProps } from '@t3n/theme';

const backgroundAnimationKeyframes = ({ theme }: ThemeProps) => keyframes`
  0% {
    background-color: ${theme.colors.shades.grey204}
  }

  50% {
    background-color: ${theme.colors.shades.grey244}
  }

  100% {
    background-color: ${theme.colors.shades.grey204}
  }
`;

// eslint-disable-next-line import/prefer-default-export
export const backgroundAnimation = css`
  animation: ${backgroundAnimationKeyframes} 2s linear infinite;
`;
