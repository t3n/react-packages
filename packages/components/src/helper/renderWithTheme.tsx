/* eslint-disable import/no-extraneous-dependencies */
import 'jest-styled-components';

import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { Theme, theme as originalTheme } from '@t3n/theme';

interface OptionalTheme {
  theme?: Theme;
}

// eslint-disable-next-line import/prefer-default-export
export const renderWithTheme = (
  ui: React.ReactElement,
  { theme, ...options }: RenderOptions & OptionalTheme
) => {
  // merge over default theme

  const mergedTheme = { ...originalTheme, ...theme };

  const Wrapper: React.FC = ({ children }) => (
    <ThemeProvider theme={mergedTheme}>{children}</ThemeProvider>
  );

  return render(ui, { wrapper: Wrapper, ...options });
};
