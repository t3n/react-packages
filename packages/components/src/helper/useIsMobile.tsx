import { Theme } from '@t3n/theme';
import { useCallback, useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';

const useIsMobile = (): boolean => {
  const [isMobile, setIsMobile] = useState(false);

  const theme: Theme = useContext(ThemeContext);
  const fontSize = window
    .getComputedStyle(window.document.body)
    .getPropertyValue('font-size');
  const mobileBreakpoint =
    parseInt(theme.breakpoints[1], 10) * parseInt(fontSize, 10);

  const testIsMobile = useCallback(() => {
    if (window.matchMedia(`(max-width: ${mobileBreakpoint}px)`).matches) {
      setIsMobile(true);
    } else {
      setIsMobile(false);
    }
  }, [mobileBreakpoint]);

  useEffect(() => {
    testIsMobile();
    window.addEventListener('resize', testIsMobile);

    return () => {
      window.removeEventListener('resize', testIsMobile);
    };
  }, [testIsMobile]);

  return isMobile;
};

export default useIsMobile;
