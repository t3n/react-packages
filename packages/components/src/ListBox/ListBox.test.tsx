import React from 'react';

import { renderWithTheme } from '../helper/renderWithTheme';
import ListBox from './ListBox';

test('ListBox Status matches Snapshot', () => {
  const { container } = renderWithTheme(
    <ListBox>Dies ist eine Listbox.</ListBox>,
    {},
  );

  expect(container).toMatchSnapshot();
});
