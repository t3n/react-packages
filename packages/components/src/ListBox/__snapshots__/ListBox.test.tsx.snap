// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBox Status matches Snapshot 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  border-color: #2a2a2a;
  border-width: 1px 2px 4px 1px;
  border-style: solid;
  border-radius: 4px;
}

@media screen and (min-width:30rem) {
  .c0 {
    padding: 16px;
  }
}

@media screen and (min-width:48rem) {
  .c0 {
    padding: 32px;
  }
}

@media screen and (min-width:61.25rem) {
  .c0 {
    padding: 32px;
  }
}

<div>
  <div
    class="c0 c1"
  >
    Dies ist eine Listbox.
  </div>
</div>
`;
